### ISSUE TITLE
Inconsistency between docs and behaviour concerning whitespace (pyparsing 2.4.7 -> 3.0 regression)

### ISSUE THREAD

**Issue by exhuma** [pyparsing/pyparsing#317]
I ran into a regression in pyparsing 3.0.1 which broke one of our parsers. The change in behaviour was *not* listed in the changelog. So it's either a regression or an oversight in the changelog 😉 

The following code worked in `pyparsing 2.4.7` but it fails in `pyparsing
3.0.1` (note the leading whitespace):

```python
from pyparsing import Word, StringStart, StringEnd

P_MTARG = (
    StringStart()
    + Word("abcde")
    + StringEnd()
)
print(P_MTARG.parseString("    aaa"))
```

In order to fix the issue, the whitespace needed to be explicitly defined:

```python
from pyparsing import Word, StringStart, Suppress, ZeroOrMore, White, StringEnd
P_MTARG = (
    StringStart()
    + Suppress(ZeroOrMore(White()))
    + Word("abcde")
    + StringEnd()
)
print(P_MTARG.parseString("    aaa"))
```

This is in conflict with the documentation, which states (emphasis mine):

> White - also similar to Word, but matches whitespace characters. **Not usually
> needed, as whitespace is implicitly ignored by pyparsing.** However, some
> grammars are whitespace-sensitive, such as those that use leading tabs or
> spaces to indicating grouping or hierarchy. (If matching on tab characters, be
> sure to call parse_with_tabs on the top-level parse element.)

See https://pyparsing-docs.readthedocs.io/en/latest/HowToUsePyparsing.html#basic-parserelement-subclasses

**Comment by exhuma** [comment]
@ptmcg I'm not at the office at the moment. I will give it a try as soon as I can.

**Comment by exhuma** [comment]
@ptmcg  I can confirm that the issue is resolved in `pyparsing>=3.0.2`. In other words, the issue _only_ exists in `pyparsing==3.0.1`

I tested 3.0.2, 3.0.3 and 3.0.4

**Comment by ptmcg** [comment]
That's great news! I'll go ahead and close this issue then.
